ONSdigital/eq-runner-mock-sds
# Support business prepop survey launch

### What is the context of this PR?
We want to be able to launch business schemas that use supplementary data. This service has been altered to serve SDS metadata and unit data back to the caller. 

A similar pattern to cache the mock unit data has been used, as in our mock CIR API. Happy to change this pattern to something else, as there isn't much SDS mock data at the moment. We could simply preload the data at startup instead, but see what you think might be best for now.

For this PR, we now do the following:

* Load the mock data from the [SDS schema examples](https://github.com/ONSdigital/sds-schema-definitions/tree/main/examples) at startup
* Dynamically create metadata and deterministically generate a `dataset_id` for each metadata chunk (similar to the mock CIR repo for its data)
* Load unit data for each survey ID with [SDS schema examples](https://github.com/ONSdigital/sds-schema-definitions/tree/main/examples). This maps internally to the dataset IDs **Note:** Survey IDs are hardcoded so we can map them to the folder names that contain the unit data.
* Identifiers and period IDs are still unused at this point.

### How to review
1. Run `make load-mock-unit-data` to pull down the example unit data from the SDS schema repo
2. Run the API
3. Make some requests to the API endpoints, e.g.: `/v1/dataset_metadata?survey_id=014&period_id=1` and `/v1/unit_data?dataset_id=d8afa921-1305-d553-d2c6-955a6db2cc2d&identifier=1`. Ensure that the data returned is as expected
4. Checkout the [Launcher PR branch](https://github.com/ONSdigital/eq-questionnaire-launcher/pull/74) and try launching various business surveys that make use of supplementary data. Ensure that the test SDS schemas still load correctly as well.

Note: As mentioned [in the comments](https://github.com/ONSdigital/eq-runner-mock-sds/pull/11#pullrequestreview-1973332853), Prodcom and BRES/BRS schemas cannot currently be tested

[Jira ticket](https://jira.ons.gov.uk/browse/ECI-1122)
